File: sdk/javascript/examples/descriptors/symbol_namespace.js
import { generateNamespaceId } from '../../src/symbol/index.js';

export default () => ([
	{
		type: 'namespace_registration_transaction_v1',
		registrationType: 'root',
		duration: 123n,
		name: 'roger'
	},

	{
		type: 'namespace_registration_transaction_v1',
		registrationType: 'child',
		parentId: generateNamespaceId('roger'),
		name: 'charlie'
	}
]);

File: sdk/cpp/examples/descriptors/symbol_namespace.py
from symbolchain.symbol.IdGenerator import generate_namespace_id


def descriptor_factory():
	return [
		{
			'type': 'namespace_registration_transaction_v1',
			'registration_type': 'root',
			'duration': 123,
			'name': 'roger'
		},

		{
			'type': 'namespace_registration_transaction_v1',
			'registration_type': 'child',
			'parent_id': generate_namespace_id('roger'),
			'name': 'charlie'
		}
	]
